### ISSUE TITLE
Express identifier as id in JSON properties

### ISSUE THREAD

**Issue by dubnemo** [OAGi/Score#1301]
For JSON Schema, and especially OpenAPI, developers have articulated their preference to have the shorter 'id'.  We do this today for our XML Schema export for OAGIS, and it would be helpful to provide this as an option via a checkbox on the expression of JSON Schema and OAS.

Examples:
Identifier -> id
Lab Request Identifier -> labRequestId

**Comment by hakjuoh** [comment]
@dubnemo @kbserm Take a look at the following cases.

<img width="947" alt="Screen Shot 2022-06-24 at 9 02 27 AM" src="https://user-images.githubusercontent.com/16102116/175541742-0f91fbfe-de4e-4ce7-840d-c040b25a1a12.png">

<img width="621" alt="Screen Shot 2022-06-24 at 9 02 58 AM" src="https://user-images.githubusercontent.com/16102116/175542373-2def015c-5dae-41cb-ae96-bff7e98038c3.png">

If we take this 'Identifier' replacement approach, we will lose the way to distinguish those terms; 'Id' and 'Identifier'.

**Comment by dubnemo** [comment]
I would argue that these examples are semantic duplicates.  This likely was a mistake when doing everything in XML Spy, and they were used in different components; e.g., noun specific BCCPs where search of components.xsd would not find the duplicate.  
Perhaps we need to identify these duplicates via a report and clean this up.

**Comment by dubnemo** [comment]
![ConvertIdentfier2id](https://user-images.githubusercontent.com/32434235/179304807-19ad3c07-c1a4-454c-80bb-c3537a0c6c11.png)

**Comment by kbserm** [comment]
@dubnemo I think it is odd to make this an option. It opens the door for interoperability problem. I think we are basically talking about revising the OAGIS BIE Serialization Specification for JSON. Another option is to make this specific to Open API because we have never published a Serialize Specification for Open API. For a long while we had a that Beta in the parenthesis behind the Open API serialization.

**Comment by dubnemo** [comment]
Key points:
1) an enterprise would be accountable for the API definition changes in their version strategy, as these would be major breaking changes.
2) if there is a need to review and change the Serialization specification, we would need to agree on what verbiage would change.

**Comment by jim-wilson-kt** [comment]
We need to create issues for duplicates. The duplicates @hakjuoh mentioned are just examples, not a list of all of them.

**Comment by dubnemo** [comment]
Usages will be created as issues in OAGIS repo
Record Set Record Id
![image](https://user-images.githubusercontent.com/32434235/180475055-f08704fc-5ea2-4a49-8be5-a63ce6880b61.png)

Record Set Record Identifier
![image](https://user-images.githubusercontent.com/32434235/180474735-4ca1adb2-824f-4659-86b8-39a8a852dfea.png)

**Comment by dubnemo** [comment]
Show. Details;  supplemental componet
![image](https://user-images.githubusercontent.com/32434235/180493382-56067f4b-e6bb-4504-8fea-b6955b03103c.png)

**Comment by hakjuoh** [comment]
@dubnemo @kbserm The following is the example result.
<img width="417" alt="Screen Shot 2022-10-06 at 11 20 17 PM" src="https://user-images.githubusercontent.com/16102116/194460989-b803ef94-63c8-4c23-9ce9-d6dbc15ea8fa.png">

Release Identifier -> releaseId
Version Identifier -> versionId
Scenario Identifier -> scenarioId
Scheme Agency Identifier -> schemeAgencyId
Document Identifier Set -> documentIdSet
Scheme Identifier -> schemeId
Identifier -> id

```yml
---
openapi: "3.0.3"
info:
  title: ""
  description: ""
  contact:
    name: ""
    url: ""
    email: "example@example.org"
  version: ""
  x-oagis-release: "10.8.3"
  x-oagis-release-date: "2022-09-30"
  x-oagis-license: ""
paths:
  /s-series/v1-0/get-bom:
    summary: ""
    description: ""
    get:
      summary: ""
      description: ""
      tags:
      - "Get BOM"
      parameters:
      - name: "id"
        in: "query"
        description: ""
        required: false
        schema:
          type: "integer"
      responses:
        "200":
          description: ""
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/getBom"
    post:
      summary: ""
      description: ""
      tags:
      - "Get BOM"
      requestBody:
        description: ""
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/getBom"
      responses:
        "200":
          description: ""
          content:
            application/json: {}
components:
  schemas:
    getBom:
      x-oagis-bie-guid: "oagis-id-6c1eb1127040434a8dee953ff5cd042f"
      x-oagis-bie-date-time: "2022-10-06T14:58:15-0400"
      x-oagis-bie-version: "v1.0"
      required:
      - "getBom"
      additionalProperties: false
      properties:
        getBom:
          type: "object"
          required:
          - "releaseId"
          - "applicationArea"
          - "dataArea"
          additionalProperties: false
          properties:
            releaseId:
              type: "string"
              format: "normalizedString"
            versionId:
              type: "string"
              format: "normalizedString"
            applicationArea:
              type: "object"
              required:
              - "creationDateTime"
              additionalProperties: false
              properties:
                creationDateTime:
                  type: "string"
                  format: "date-time"
                scenarioId:
                  type: "object"
                  required:
                  - "content"
                  additionalProperties: false
                  properties:
                    content:
                      type: "string"
                      format: "normalizedString"
                    schemeAgencyId:
                      type: "string"
                      format: "token"
            dataArea:
              type: "object"
              required:
              - "get"
              - "bom"
              additionalProperties: false
              properties:
                get:
                  type: "object"
                  required:
                  - "expression"
                  additionalProperties: false
                  properties:
                    expression:
                      type: "array"
                      minItems: 1
                      items:
                        type: "string"
                        format: "token"
                bom:
                  type: "array"
                  minItems: 1
                  items:
                    type: "object"
                    additionalProperties: false
                    properties:
                      bomHeader:
                        type: "object"
                        additionalProperties: false
                        properties:
                          documentIdSet:
                            type: "array"
                            items:
                              type: "object"
                              required:
                              - "id"
                              additionalProperties: false
                              properties:
                                schemeId:
                                  type: "string"
                                  format: "normalizedString"
                                id:
                                  type: "array"
                                  minItems: 1
                                  items:
                                    type: "string"
                                    format: "normalizedString"
```

**Comment by jim-wilson-kt** [comment]
I like it!

**Comment by dubnemo** [comment]
Done, deserves a patch release!!

**Comment by dubnemo** [comment]
@hakjuoh @kbserm 
I see you removed this from 2,4 milestone.  Was there an issue?

**Comment by hakjuoh** [comment]
> @hakjuoh @kbserm I see you removed this from 2,4 milestone. Was there an issue? We had consensus to move forward, and the above looked great. Can we get this in a 2.4.1 release?

It was removed from 2.4 milestone by mistake. I checked that this issue was solved in 2.4 milestone.
